### PR TITLE
Add a publishToS3 method

### PR DESCRIPTION
**Pull request by rowanmanning**
Currently only available in the JavaScript API. I'll add the CLI in a separate PR.

### CODE REVIEW

File: lib/origami-image-set-tools.js [hunk 1]
```diff
@@ -16,6 +19,8 @@ module.exports = class OrigamiImageSetTools {
 	constructor(options) {
 		this.options = defaults({}, options, module.exports.defaults);
 		this.log = this.options.log;
+		this.scheme = this.options.scheme;
+		this.version = (semver.valid(this.options.version) ? this.options.version : 'v0.0.0');
```
Comment (JakeChampion) [inline lib/origami-image-set-tools.js]: What do you think about having the `else` case pulling it's value from `module.exports.defaults.version`?

File: lib/origami-image-set-tools.js [hunk 2]
```diff
@@ -51,10 +56,61 @@ module.exports = class OrigamiImageSetTools {
...
-			const filePath = path.join(this.options.baseDirectory, 'imageset.json');
-			const fileContents = JSON.stringify(imageSetManifest, null, '\t');
-			return fs.writeFile(filePath, fileContents);
+			return Promise.all(imageSetManifest.images.map(image => {
+				const fullPath = path.resolve(this.options.baseDirectory, image.path);
+				const s3Path = `${s3BasePath}/${image.name}.${image.extension}`;
+				const s3Config = {
+					ACL: 'public-read',
+					Body: fs.createReadStream(fullPath),
+					ContentType: mime.lookup(fullPath),
```
Comment (JakeChampion) [inline lib/origami-image-set-tools.js]: `mime.lookup` can return `false`, should we let that happen or should we have a whitelist of image mime types?

Comment (rowanmanning) [inline lib/origami-image-set-tools.js]: yep there's already a whitelist – the function that builds the manifest uses [this array](https://github.com/Financial-Times/origami-image-set-tools/blob/1388562ded7b78932eb966a538e312e7615b4c92/lib/origami-image-set-tools.js#L137) to filter files.

File: lib/origami-image-set-tools.js (hunk 1 above)
Comment (rowanmanning) [inline lib/origami-image-set-tools.js]: Good spot, done

**Pull request merged**